app/scripts/background.js

```diff
@@ -367,17 +361,38 @@ function maybeDetectPhishing(theController) {
   );
 }
 
+function stringifyError(error) {
+  return JSON.stringify({
+    message: error.message,
+    name: error.name,
+    stack: error.stack,
+  });
+}
+
 // These are set after initialization
 let connectRemote;
 let connectExternalExtension;
 let connectExternalCaip;
 
 browser.runtime.onConnect.addListener(async (...args) => {
   // Queue up connection attempts here, waiting until after initialization
-  await isInitialized;
+  try {
+    await isInitialized;
+    connectRemote(...args);
+  } catch (error) {
+    const port = args[0];
+
+    const _state = await localStore.get();
+    sentry?.captureException(error);
+
+    port.postMessage({
+      target: 'ui',
+      error: stringifyError(error),
+      metamaskState: JSON.stringify(_state),
+    });
+  }
 
   // This is set in `setupController`, which is called as part of initialization
```

Comment:
Nit: This comment should probably be moved to follow `connectRemote`. It was meant to explain why we don't need to worry about it being undefined.

Reply:
fix in https://github.com/MetaMask/metamask-extension/pull/31404/commits/c268a0e0f3a2f52f22c4600eafe60f0056ed2d71